tests/lib/nested.sh

```diff
@@ -24,15 +24,35 @@ NESTED_UBUNTU_IMAGE_SNAPPY_FORCE_SAS_URL="${NESTED_UBUNTU_IMAGE_SNAPPY_FORCE_SAS
 
 nested_wait_for_ssh() {
     # TODO:UC20: the retry count should be lowered to something more reasonable.
-    nested_retry_until_success 800 1 "true"
+    local retry=800
+    local wait=1
+
+    until nested_exec "true"; do
+        retry=$(( retry - 1 ))
+        if [ $retry -le 0 ]; then
+            echo "Timed out waiting for command 'true' to succeed. Aborting!"
+            return 1
+        fi
+        sleep "$wait"
+    done
 }
 
 nested_wait_for_no_ssh() {
-    nested_retry_while_success 200 1 "true"
+    local retry=200
+    local wait=1
+
+    while nested_exec "true"; do
```

Comment:
couldn't this be replaced with the same sort of implementation we have now for `nested_wait_for_snap_command`?

Reply:
In a following PR it will be updated, this is the first part, the we have the 10500 and finally a new one which move/update all those functions to the new tool.